service/src/main/java/bio/terra/workspace/app/controller/ResourceApiController.java

```diff
@@ -111,6 +116,54 @@ public ResponseEntity<ApiResourceDescription> getResource(UUID workspaceUuid, UU
     return new ResponseEntity<>(apiResourceDescription, HttpStatus.OK);
   }
 
+  @WithSpan
+  @Override
+  public ResponseEntity<ApiResourceDescription> getResourceByName(
+      UUID workspaceUuid, String resourceName) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+
+    WsmResource wsmResource =
+        validateWorkspaceOrResourceReadAccess(userRequest, workspaceUuid, resourceName);
+
+    ApiResourceDescription apiResourceDescription = this.makeApiResourceDescription(wsmResource);
+    return new ResponseEntity<>(apiResourceDescription, HttpStatus.OK);
+  }
+
+  private WsmResource validateWorkspaceOrResourceReadAccess(
+      AuthenticatedUserRequest userRequest, UUID workspaceUuid, String resourceName) {
+    String readAction = SamConstants.SamControlledResourceActions.READ_ACTION;
+
+    try {
+      workspaceService.validateWorkspaceAndAction(userRequest, workspaceUuid, readAction);
+      return resourceService.getResourceByName(workspaceUuid, resourceName);
+    } catch (ForbiddenException workspaceException) {
+      WsmResource resource = resourceService.getResourceByName(workspaceUuid, resourceName);
+      switch (resource.getStewardshipType()) {
+        case CONTROLLED -> {
+          try {
+            return controlledResourceMetadataManager.validateControlledResourceAndAction(
+                userRequest, workspaceUuid, resource.getResourceId(), readAction);
+          } catch (ForbiddenException resourceException) {
+            // if no access, throw original exception to avoid leaking information on resource
+            throw workspaceException;
+          }
+        }
+        case REFERENCED -> {
```

Comment:
I'm not sure this is exactly what we want for a referenced resource. Checking access on the referenced resource involves a permissions check on the underlying cloud resource that's referenced. If a user doesn't have access to a workspace, then access to the underlying resource that's referenced shouldn't grant access to read the reference itself too IMO. 

Not really sure what the downside of allowing the access would be, but still best prevent it